src/server/pfs/server/driver_new_storage.go

```diff
@@ -54,29 +57,24 @@ func (d *driver) finishCommitNewStorageLayer(txnCtx *txnenv.TransactionContext,
 	if description != "" {
 		commitInfo.Description = description
 	}
-	// Close in-memory file set (serializes in-memory part).
-	if err := d.fs.Close(); err != nil {
+	// Compact the commit changes into a diff file set.
+	commitPath := path.Join(commit.Repo.Name, commit.ID)
+	if err := d.compact(context.Background(), path.Join(commitPath, fileset.Diff), []string{commitPath}); err != nil {
```

Comment:
Why are these calls using `context.Background` rather than propagating the context from the `txnCtx`?

Reply:
I was a little unsure about what context to get from the transaction env (`ClientContext` vs `Client.Ctx()`). I talked to @Tryneus and he suggested `Client.Ctx()`.